src/js/core/util/scrolling.js

```diff
@@ -309,28 +369,51 @@
 				}
 			}
 
+			function getSnapSize(index) {
+				if (snapPoints) {
+					return Math.abs(snapPoints[previousIndex].position - snapPoints[index].position);
+				} else {
+					return snapSize;
+				}
+			}
+
 			/**
 			 * Handler for rotary event
 			 * @param {Event} event
 			 */
 			function rotary(event) {
 				var eventDirection = event.detail && event.detail.direction;
 
+				previousIndex = currentIndex;
+
 				// update position by snapSize
 				if (eventDirection === "CW") {
-					moveToPosition -= snapSize || 50;
+					currentIndex++;
+					if (snapPoints && currentIndex >= snapPoints.length) {
+						currentIndex = snapPoints.length - 1;
+					}
+					snapSize = -1 * getSnapSize(currentIndex);
+
 				} else {
-					moveToPosition += snapSize || 50;
+					currentIndex--;
+					if (snapPoints && currentIndex < 0) {
+						currentIndex = 0;
+					}
+					snapSize = getSnapSize(currentIndex);
 				}
+
+				moveToPosition += snapSize;
+
 				if (snapSize) {
-					moveToPosition = snapSize * round(moveToPosition / snapSize);
+					//moveToPosition = snapSize * round(moveToPosition / snapSize);
```

Comment:
I think it should not be removed. It will break compatibility with previous solution where snapSize was constant for all elements.

I think the whole function could have a check at the begining if use snapSize or snapPoints

Reply:
Done.